BackEnd/src/test/kotlin/quest/marketstack/tradingapp/closedtrades/service/ClosedOpenTradeServiceTest.kt

```diff
@@ -0,0 +1,41 @@
+package quest.marketstack.tradingapp.closedtrades.service
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Test
+import quest.marketstack.tradingapp.domain.closedtrades.ClosedTradeDataSource
+import quest.marketstack.tradingapp.model.ClosedTrade
+import quest.marketstack.tradingapp.service.closedtrades.ClosedTradeService
+
+class ClosedOpenTradeServiceTest {
+    private val dataSource: ClosedTradeDataSource = mockk()
+
+    private val tradeService = ClosedTradeService(dataSource)
+
+    @Test
+    fun `getTrades() should call it's data source to retrieve closed trades`() {
+        val testClosedTrades = getTestClosedTrades()
+
+        every {
+            dataSource.retrieveTrades()
+        } returns testClosedTrades
+
+        val trades = tradeService.getTrades()
+
+        assertNotNull(trades)
+        verify(exactly = 1) {
+            dataSource.retrieveTrades()
+        }
+    }
+
+    private fun getTestClosedTrade(): ClosedTrade =
+        ClosedTrade(
+            id = "1",
+            tradeExecs = mutableListOf(),
+            shortLong = true,
+        )
+
+    private fun getTestClosedTrades(): Collection<ClosedTrade> = listOf(getTestClosedTrade())
+}
```

Comment:
I noticed the way you were writing unit tests wasn't really accurate/testing the correct things. This is an example I want you to follow, and I want you to go through every single test file you have, delete the test file you wrote, and rewrite them based on this example. Here's a blueprint you should follow when writing a test class:

1. When creating a test class for a file, whether it be service, controller, anything, you should be testing every single function in the class. Let's say you have a class with 3 functions, A, B, and C, and each of these 3 functions have two possible outcomes. There should be a test for each outcome per function, therefore there should be 6 tests. In this scenario, you only have two get functions that don't perform any logic (one possible outcome each), so you should only need two tests.
2. Each test should follow the following format:
    a. At the top, define any constants you might need. In this case, that's `testClosedTrades`. In order to avoid writing repetitive code, if the constant you're making is an object (not like an int or a String), make a private helper function at the bottom that just returns the object (this is the `getTestClosedTrade()` function I made.) Notice how I'm able to now reuse that function in  `getTestClosedTrades()`, and therefore there's no repetitive code.
    
   b. Since you're testing the service class, and it's a _unit_ test, all you care about is that the function you're testing is going through the correct steps. You don't care about what the values of variables are, you only care that the correct steps occur. So, if the function you're testing (let's call it function A) is calling another function (let's call it function B), you should be simulating the result of function B. You don't need to call it for real, but that's where the `every { }` block comes in. That block just says that "every" time that function is called, return testClosedTrades.
   
   Note here: You put `mockk(relaxed = true)` to avoid having to do that. **_Don't do that_**
   
   c. Once you've defined all of the `every` blocks that are needed (if you're missing one, you'll get an error that says something like "No answer for *********"), you can go ahead and make the call that you're testing, in this case, `tradeService.getTrades()`. Assign that to a variable so you can use it for validation (line 25)

   d. Using the result of the function call that you assigned to a variable in step 2c (`val trades`), perform any assertions, which you did already in line 27, that's good!

   e. Final step, and this is the counterpart to step 2b, and operates almost exactly the same way. Step 2b is using the `every {}` block to provide "dummy" values for each call to another function that the function you're testing makes. Here, you want to use the `verify {}` block to make sure that those functions are actually called. For example, line 28, `verify(exactly = 1) { dataSource.retrieveTrades() }` is making sure that `dataSource.retrieveTrades()` is called once and only once.